src/ir_York.cpp

```diff
@@ -0,0 +1,358 @@
+    default: return stdAc::fanspeed_t::kAuto;
+  }
+}
+
+/// Set the temperature.
+/// @param[in] degrees The temperature in degrees celsius.
+void IRYork::setTemp(const uint8_t degrees) {
+  uint8_t temp = std::max(kYorkMinTemp, degrees);
+  temp = std::min(kYorkMaxTemp, temp);
+  _.Temp = (temp * 4);
+}
+
+/// Get the current temperature setting.
+/// @return Get current setting for temp. in degrees celsius.
+uint8_t IRYork::getTemp(void) const {
+  return _.Temp / 4;
+}
+
+/// Set the On Timer value of the A/C.
+/// @param[in] nr_of_mins The number of minutes the timer should be.
+/// @note The timer time only has a resolution of 10 mins.
+/// @note Setting the On Timer active will cancel the Sleep timer/setting.
+void IRYork::setOnTimer(const uint16_t nr_of_mins) {
+  // TODO: York AC unit uses 10 minutes increments from now
+  _.OnTime = nr_of_mins;
+}
+
+/// Set the Off Timer value of the A/C.
+/// @param[in] nr_of_mins The number of minutes the timer should be.
+/// @note The timer time only has a resolution of 10 mins.
+/// @note Setting the Off Timer active will cancel the Sleep timer/setting.
+void IRYork::setOffTimer(const uint16_t nr_of_mins) {
+  // TODO: York AC unit uses 10 minutes increments from now
+  _.OffTime = nr_of_mins;
+}
+
+
+/// Get the On Timer setting of the A/C.
+/// @return The Nr. of minutes the On Timer is set for.
+uint16_t IRYork::getOnTimer(void) const {
+  // TODO: York AC unit uses 10 minutes increments from now
+  return _.OnTime;
+}
+
+/// Get the Off Timer setting of the A/C.
+/// @return The Nr. of minutes the Off Timer is set for.
+/// @note Sleep & Off Timer share the same timer.
+uint16_t IRYork::getOffTimer(void) const {
+  // TODO: York AC unit uses 10 minutes increments from now
+  return _.OffTime;
+}
+
+/// Convert the current internal state into its stdAc::state_t equivalent.
+/// @param[in] prev Ptr to the previous state if required.
+/// @return The stdAc equivalent of the native settings.
+stdAc::state_t IRYork::toCommon(const stdAc::state_t *prev) const {
+  //TODO: all copy/pasted
+  stdAc::state_t result{};
+  // Start with the previous state if given it.
+  if (prev != NULL) {
+    result = *prev;
+  } else {
+    // Set defaults for non-zero values that are not implicitly set for when
+    // there is no previous state.
+    // e.g. Any setting that toggles should probably go here.
+    result.power = false;
+  }
+  result.protocol = decode_type_t::YORK;
+  result.mode = toCommonMode(_.Mode);
+  result.celsius = true;
+  result.degrees = getTemp();
+  result.fanspeed = toCommonFanSpeed(_.Fan);
+  result.swingv = _.SwingV ? stdAc::swingv_t::kAuto : stdAc::swingv_t::kOff;
+  // Not supported.
+  result.model = -1;
+  result.turbo = false;
+  result.swingh = stdAc::swingh_t::kOff;
+  result.light = false;
+  result.filter = false;
+  result.econo = false;
+  result.quiet = false;
+  result.clean = false;
+  result.beep = false;
+  result.clock = -1;
```

Comment:
What happened to Sleep?

Reply:
See the updated top comment (meanwhile I added sleep as unsupported with a TODO)